### PR TITLE
[api] fix maintenance_statistics response when a project is not found

### PR DESCRIPTION
**Pull request by evanrolfe**
Fixes issue https://github.com/openSUSE/open-build-service/issues/2696

### CODE REVIEW

File: src/api/app/controllers/statistics/maintenance_statistics_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class MaintenanceStatisticsController < ApplicationController
     skip_before_action :extract_user
 
     def index
-      @project = Project.find_by(name: params[:project])
+      @project = Project.find_by!(name: params[:project])
```
Comment (hennevogel) [inline src/api/app/controllers/statistics/maintenance_statistics_controller.rb]: I'm not sure this applies here but isn't that circumventing all the access checks? @adrianschroeter?

Comment (evanrolfe) [inline src/api/app/controllers/statistics/maintenance_statistics_controller.rb]: actually this is a good point to make, it reminds me that @ChrisBr and I discussed that this endpoint is only showing publicly available information so doesn't need to have an authentication. @hennevogel @adrianschroeter what do you guys think about removing the authorisation checks from this endpoint?

**Comment by adrianschroeter** [comment]
On Donnerstag, 16. Februar 2017, 08:33:47 CET wrote Evan Rolfe:
> evanrolfe commented on this pull request.
> 
> 
> 
> > @@ -3,7 +3,7 @@ class MaintenanceStatisticsController < ApplicationController
>      skip_before_action :extract_user
>  
>      def index
> -      @project = Project.find_by(name: params[:project])
> +      @project = Project.find_by!(name: params[:project])
> 
> actually this is a good point to make, it reminds me that @ChrisBr and I discussed that this endpoint is only showing publicly available information so doesn't need to have an authentication. @hennevogel @adrianschroeter what do you guys think about removing the authorisation checks from this endpoint?

we need to handle the situation that the project is hidden, but the user still has
access to it. 

The Project.find method will fail for remote projects btw.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg) 
 
Maxfeldstraße 5                         
90409 Nürnberg 
Germany

File: src/api/app/controllers/statistics/maintenance_statistics_controller.rb (hunk 1 above)
Comment (hennevogel) [inline src/api/app/controllers/statistics/maintenance_statistics_controller.rb]: authentication != authorization :-) This does not need authentication (should be public) but it needs authorization (should not show hidden project).  I would suggest you use Project.find_by_name.

**NOTE** this is not a built in magic finder but something self implemented.

**Comment by ChrisBr** [comment]
@adrianschroeter does the maintenance statistic make sense for remote projects at all? Just wondering as we use issues, history elements etc to calculate the statistics.

**Comment by bgeuken** [comment]
@evanrolfe might make sense to go directly to Adrian and hash this out.

**Comment by evanrolfe** [comment]
@adrianschroeter @hennevogel after discussing this with bjoern we have decided to create a new card for the task of making the maintenance statistics work for remote projects:
https://trello.com/c/X3kff6Up/183-make-maintenance-statistics-work-for-remote-projects

Please can you review the existing code in this PR, just to make sure there authorization on this api endpoint. Thanks

**Pull request merged**